cmd/soroban-rpc/internal/methods/get_events.go

```diff
@@ -17,15 +18,65 @@ import (
 	"github.com/stellar/soroban-tools/cmd/soroban-rpc/internal/events"
 )
 
+type eventTypeSet map[string]interface{}
+
+func (e eventTypeSet) valid() error {
+	for key := range e {
+		switch key {
+		case EventTypeSystem, EventTypeContract, EventTypeDiagnostic:
+			// ok
+		default:
+			return errors.New("if set, type must be either 'system', 'contract' or 'diagnostic'")
+		}
+	}
+	return nil
+}
+
+func (e *eventTypeSet) UnmarshalJSON(data []byte) error {
+	if len(data) == 0 {
+		*e = map[string]interface{}{}
+		return nil
+	}
+	var joined string
+	if err := json.Unmarshal(data, &joined); err != nil {
+		return err
+	}
+	*e = map[string]interface{}{}
+	if len(joined) == 0 {
+		return nil
+	}
+	for _, key := range strings.Split(joined, ",") {
+		(*e)[key] = nil
+	}
+	return nil
+}
+
+func (e eventTypeSet) MarshalJSON() ([]byte, error) {
+	var keys []string
+	for key := range e {
+		keys = append(keys, key)
+	}
+	return json.Marshal(strings.Join(keys, ","))
+}
+
+func (e eventTypeSet) matches(event xdr.ContractEvent) bool {
+	if len(e) == 0 {
+		return true
+	}
+	_, ok := e[eventTypeFromXDR[event.Type]]
+	return ok
+}
+
 type EventInfo struct {
-	EventType      string         `json:"type"`
-	Ledger         int32          `json:"ledger,string"`
-	LedgerClosedAt string         `json:"ledgerClosedAt"`
-	ContractID     string         `json:"contractId"`
-	ID             string         `json:"id"`
-	PagingToken    string         `json:"pagingToken"`
-	Topic          []string       `json:"topic"`
-	Value          EventInfoValue `json:"value"`
+	EventType                  string         `json:"type"`
+	Ledger                     int32          `json:"ledger,string"`
+	LedgerClosedAt             string         `json:"ledgerClosedAt"`
+	ContractID                 string         `json:"contractId"`
+	ID                         string         `json:"id"`
+	PagingToken                string         `json:"pagingToken"`
+	Topic                      []string       `json:"topic"`
+	Value                      EventInfoValue `json:"value"`
+	FromSuccessfulContractCall bool           `json:"fromSuccessfulContractCall"`
```

Comment:
What about `InSuccessfulContractCall` to match the xdr naming?
```suggestion
	InSuccessfulContractCall bool           `json:"inSuccessfulContractCall"`
```